sys/event/timeout.c

```diff
@@ -10,7 +10,7 @@
 
 #include "kernel_defines.h"
 #include "event/timeout.h"
-#if IS_USED(MODULE_ZTIMER_USEC)
+#if IS_USED(MODULE_ZTIMER64_USEC) && !IS_USED(MODULE_XTIMER)
```

Comment:
this forces ztimer64_usec for every thing that uses event time out (even though event time out clearly just needs ztimer usec)

Reply:
... unless MODULE_EVENT_TIMEOUT_ZTIMER is used.

The "old" event_timeout depends on xtimer semantics. users of that block low power mode, whether they use xtimer_compat based on ztimer or ztimer64 doesn't matter. pulling the clock under them breaks them in any case.

xtimer_compat doesn't solve the lack of power management awareness of xtimer-using code, it just allows using ztimer as base timer.

Reply:
if something is using just the event timeout interface it does not depend on xtimer semantics -> only if it uses xtimer it implies xtimer semantics
in makefile.dep there is one (direct) user event timeout "asymcute" that does not other wise specify a timer implementation and it does not use any xtimer

Reply:
now I see your point I think. There's no reason to use a ztimer64 unless ztimer_xtimer_compat is used, if that needs one.
But in that case, event timeout with using the compat header is not compatible anymore with event timeout just using ztimer32.

Can't we just remove xtimer as backend timer here? keep the old api (without specifying a clock -> ztimer_usec is used). (out of scope I guess).

Reply:
in ecence event timeout dos not depend on xtimer i am not sure if this should be kept working on an non xtimer configuration but using xtimer_compat64 for its implementation is realy super plus ungood

Reply:
If no ztimer-generation code is using event_timer, why is compat64 worse compared to compat32?

(*if* ztimer generation code is using event_timer, it is using ztimer_usec(32) anyways)

Reply:
all users of event timeout are either ztimer(32) enabled or use no other timer (asymcute) -> using ztimer64 is ++ungood
<details>
I see i need to get better at getting my thoughts into written form
</details>